server/utils/uniq.go

```diff
@@ -0,0 +1,11 @@
+package utils
+
+//AppendIfMissing appends to an array only uniq elements
+func AppendIfMissing(slice []string, i string) []string {
```

Comment:
It looks like all the usages of this func are calls from loops, which is `O(n^2)` overall. It would be more efficient to build `map[string]struct{}`s along the way, then iterate once at the end to make the slice if necessary.